File: ml_service/routes/preferences.py
from fastapi import APIRouter, HTTPException
from pydantic import BaseModel
from typing import Optional

router = APIRouter()

class UserPreferences(BaseModel):
    theme: str = 'dark'
    sidebarOpen: bool = True
    codeFont: str = 'JetBrains Mono'
    fontSize: int = 14
    autoComplete: bool = True
    notifications: bool = True

# In-memory storage for preferences (replace with database in production)
user_preferences = {}

@router.get("/preferences")
async def get_preferences():
    # In a real app, you would get this from a database based on the user's ID
    return user_preferences.get("default", UserPreferences().dict())

@router.post("/preferences/update")
async def update_preferences(preferences: UserPreferences):
    if preferences.theme not in ['dark', 'light']:
        raise HTTPException(status_code=400, detail="Invalid theme")
    
    if preferences.fontSize < 8 or preferences.fontSize > 32:
        raise HTTPException(status_code=400, detail="Font size must be between 8 and 32")
    
    # In a real app, you would save this to a database with the user's ID
    user_preferences["default"] = preferences.dict()
    return user_preferences["default"]

File: src/api/preferences.ts
import { API_ENDPOINTS } from './config';
import { apiClient } from './client';

export interface UserPreferences {
  theme: 'dark' | 'light';
  sidebarOpen: boolean;
  codeFont: string;
  fontSize: number;
  autoComplete: boolean;
  notifications: boolean;
}

const defaultPreferences: UserPreferences = {
  theme: 'dark',
  sidebarOpen: true,
  codeFont: 'JetBrains Mono',
  fontSize: 14,
  autoComplete: true,
  notifications: true,
};

export const preferencesApi = {
  get: async () => {
    try {
      const response = await apiClient.get<UserPreferences>(API_ENDPOINTS.preferences.get);
      return response.data;
    } catch (error) {
      console.warn('Failed to load preferences, using defaults:', error);
      return defaultPreferences;
    }
  },

  update: async (preferences: Partial<UserPreferences>) => {
    const response = await apiClient.post<UserPreferences>(API_ENDPOINTS.preferences.update, preferences);
    return response.data;
  }
};
